pkg/plugins/plugin_manager.go

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package plugins
+
+import (
+	"sync"
+
+	"github.com/dragonflyoss/Dragonfly/supernode/config"
+)
+
+// NewManager creates a default plugin manager instant.
+func NewManager() Manager {
+	return &managerIml{
+		builders: NewRepository(),
+		plugins:  NewRepository(),
+	}
+}
+
+// NewRepository creates a default repository instant.
+func NewRepository() Repository {
+	return &repositoryIml{
+		repos: make(map[config.PluginType]*sync.Map),
+	}
+}
+
+// Manager manages all plugin builders and plugin instants.
+type Manager interface {
```

Comment:
I had a feeling that these functions might be better to include `error` as return value.